lib/services/ios-debug-service.ts

```diff
@@ -158,13 +158,13 @@ export class IOSDebugService extends DebugServiceBase implements IPlatformDebugS
 
 		const iOSEmulatorService = <Mobile.IiOSSimulatorService>this.$iOSEmulatorServices;
 		await iOSEmulatorService.postDarwinNotification(attachRequestMessage);
-		await this.connectToApplicationOnEmulator(debugData.deviceIdentifier);
+		await this.waitForBackendPortToBeOpened(debugData.deviceIdentifier);
 		return result;
 	}
 
-	private async connectToApplicationOnEmulator(deviceIdentifier: string): Promise<void> {
-		const socket = await this.$iOSEmulatorServices.connectToPort({ port: inspectorBackendPort });
-		if (!socket) {
+	private async waitForBackendPortToBeOpened(deviceIdentifier: string): Promise<void> {
+		const portListens = await this.$net.waitForPortToListen(inspectorBackendPort, 10000, 200);
```

Comment:
Introducing an object instead of passing three separate values here would increase readability a lot IMO